packages/flutter_tools/lib/src/protocol_discovery.dart

```diff
@@ -60,10 +61,14 @@ class ProtocolDiscovery {
 
   void _handleLine(String line) {
     Uri uri;
+
+    final RegExp regExp = new RegExp(r'[a-z]+:\/\/[^ \n]*');
```

Comment:
I don't think you need to escape the slashes, and you should require something after them.  Also, it shouldn't be necessary to use the regexp _and_ the `indexOf()`, so let's combine them.  That would yield a regexp of:

`r'http://[^ \n]+'`

Reply:
Note that this also means you can nuke the `_prefix` instance variable now too.